### DISCUSSION TITLE
3.5.0

### DISCUSSION THREAD

**Discussion by Micke-K** [Micke-K/IntuneManagement#47] [Announcements]
**New features**

- **Automatic update check**
  The app will check GitHub at start-up if there is a new version available
  This can be disabled in settings

- **Use PowerShell 5**
  Command files will now use PowerShell 5 (-version 5 in the command line)
  This is based on [Issue 44](https://github.com/Micke-K/IntuneManagement/issues/44)

- **Documentation**
  - New Word settings: Table text style and table caption location
  This is based on an additional request in [Issue 37](https://github.com/Micke-K/IntuneManagement/issues/37)
  - Terms of Use info when documenting Conditional Access
  - Added documentation support for Terms of Use
  - Added additional support for offline documentation
  **Note:** Offline is defined as documenting an exported folder while logged in to another tenant. 
  If logged in to the same tenant as the exported folder, "online" documentation will be used
  - Changed the layout for the assignment table on Win32 Applications. There were too many columns so additional info is changed to a table in the value column
  - Filter / Filter Mode column headers are now set from language files

- **Export/Import**
  - Users in Conditional Access are now added to the Migration Table
  This is so the user IDs can be translated during Offline documentation
  - Referenced settings are now included in the export
  This is to support referenced settings during import, copy and offline documentation (Certs on VPN profiles etc.)
  These properties are named #CustomRef_*PropertyName* in the json file
  **Note:** This might cause export/copy to take longer once every second week since it requires the MetaData XML for Graph to be downloaded.
  This feature can be turned off by unchecking 'Resolve reference info' in Settings

- **Copy**
  - New dialog when copying an object. Description can now be changed during the copy

- **Authentication**
  - Full authentication support for US Government and China clouds
  This requires that 'Show Azure AD login menu' is enabled in Settings
  - Consent can be requested for missing permissions. This can be triggered via the 'Request Consent' link in the user profile info
  - New version of MSAL.DLL, version 4.42.1
  - Object types with only Read permissions are now supported. These will be orange in the menu
  Buttons like Import and Delete will still be available but they will not work

- **List objects**
  - IsAssigned column is added to objects that supports it (property on the Graph object)
  - Enable 'Expand assignments' in Settings to include Assignments when getting a full list of objects from Graph
  This can be used for adding Custom columns based on assignment info
  It is also used for setting the IsAssigned column for objects that doesn't have the info in Graph
  This is based on [Issue 30](https://github.com/Micke-K/IntuneManagement/issues/30)
  - Apps can be filtered in the request
  If there are more than 1000 applications in the environment, the filter box can be used to return only matched items
  Enter the filter in the text box and press the Refresh button. Clear the filter box and click Refresh to reload other objects
  This is based on [Issue 28](https://github.com/Micke-K/IntuneManagement/issues/28)

**Fixes**

- **Documentation**

  - Fixed bug in *Conditional Access* documentation that caused some Grant information to be excluded from documentation
  - Fixed missing properties when documenting *Device restrictions (Windows 10 Team)* profiles 
  - Fixed some Offline Documentation issues
  Get dependency info from exported folders instead of Graph
  Offline documentation is not 100% fully supported yet. Dependency applications for Win32 apps are not included in this version
  and there might be more properties missing. Please report anything missing for offline documentation to [Issue 37](https://github.com/Micke-K/IntuneManagement/issues/37)
  **Note** Offline documentation will always require online access. Some information like language text, Azure roles, Mobile apps etc. will use Graph API

- **Authentication**
  - First login with last used account could fail if the user domain was changed after the initial token was cached

<hr /><em>This discussion was created from the release <a href='https://github.com/Micke-K/IntuneManagement/releases/tag/3.5.0'>3.5.0</a>.</em>

**Comment by endersstory** [comment]
Hello, 

Thank you for your work on this amazing tool. Has made managing my test and production tenants so much easier. I can't believe you're making it better!

Thanks for fixing the authentication issue as well. I didn't want to post that as an issue as, kind of expected the tool to mess up when using different accounts.

**Reply by Micke-K** [reply]
Thank you for the kind words and the support!

What was the issue you had with the login?

**Reply by endersstory** [reply]
The accounts I use the password changes after the specified duration. When I would leave the IntuneManagement app running, the password expires, and session times out. I could not login with the new password with the existing session of the app.

Fix is simple.
1. exit all current Intune expired sessions completely
2. restart the app.
